src/types/lib.ts

```diff
@@ -77,6 +77,19 @@ export type InvocationsDetails = {
 
 export type InvocationsDetailsWithNonce = InvocationsDetails & { nonce: BigNumberish };
 
+export type TransactionBulk = { type: TransactionType } & (
```

Comment:
@irisdv why are you calling it "Bulk" if it's a single object? I would move the wrapping in an Array to the type to justify it's "bulk" naming and change function signature to accept TransactionBulk rather than Array<TransactionBulk>